fluent/syntax/parser.py

```diff
@@ -349,13 +349,14 @@ def get_number(self, ps):
     @with_span
     def get_pattern(self, ps):
         elements = []
-        ps.skip_ws()
+        ps.skip_inline_ws()
 
         while ps.current():
             ch = ps.current()
 
             if ch == '\n':
-                break
+                if not ps.is_peek_next_line_pattern():
+                    break
```

Comment:
Thanks. After some more thought I decided to leave the second `is_peek_next_line_pattern` in, with a comment explaining the end-condition of `get_pattern`'s loop.

Checking for length of the value of the last `TextElement` would mean that we create the element, add spans to it in `with_span` and then discard it. Returning `None` would in turn create a precedent where we need to check the truthiness of nodes returned from `get_` methods, at least in `with_span`.

Another reason to not optimize this code wrt. duplication right now is that a lot of it might change when we fix #15. If it then turns out it's easier to return `None` after all, we can add it.